src/tests/functional/plugin/gna/shared_tests_instances/single_layer_tests/activation.cpp

```diff
@@ -11,12 +11,32 @@ namespace LayerTestsDefinitions {
 class ActivationLayerGNATest : public ActivationLayerTest {
 protected:
     void SetUp() override {
-        ActivationLayerTest::SetUp();
+        InferenceEngine::Precision netPrecision;
+        std::pair<std::vector<size_t>, std::vector<size_t>> shapes;
+        std::pair<ngraph::helpers::ActivationTypes, std::vector<float>> activationDecl;
+        std::tie(activationDecl, netPrecision, inPrc, outPrc, inLayout, outLayout, shapes, targetDevice) = GetParam();
+
+        auto inputShape = shapes.first;
+        const std::size_t inputDim = std::accumulate(inputShape.begin(), inputShape.end(), 1, std::multiplies<size_t>());
+        std::vector<size_t> inputDims{1, inputDim};
+        activationType = activationDecl.first;
+        auto constantsValue = activationDecl.second;
+        auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
+        auto params = ngraph::builder::makeParams(ngPrc, {inputDims});
+        params[0]->set_friendly_name("Input");
+
         // TODO: remove after integer inference output support
-        auto ngPrc = function->get_parameters()[0]->get_element_type().get_type_name();
-        if (ngPrc == "u8" || ngPrc == "i16") {
+        if (ngPrc == ngraph::element::u8 || ngPrc == ngraph::element::i16) {
             threshold = 1.0;
         }
+
+        auto inputReshapePattern = std::make_shared<ngraph::opset1::Constant>(ngraph::element::i64, ngraph::Shape{inputShape.size()}, inputShape);
+        auto inputReshape = std::make_shared<ngraph::opset1::Reshape>(params[0], inputReshapePattern, false);
+        auto activation = ngraph::builder::makeActivation(inputReshape, ngPrc, activationType, shapes.second, constantsValue);
+        auto outputReshapePattern = std::make_shared<ngraph::opset1::Constant>(ngraph::element::i64, ngraph::Shape{2}, inputDims);
+        auto outputReshape = std::make_shared<ngraph::opset1::Reshape>(activation, outputReshapePattern, false);
```

Comment:
Use `const` whenever possible.

Reply:
We should rewrite half of the code :-)

Reply:
My aim is to put attention on this aspect in new or being changed code. I'm not asking to rewrite PR unrelated pieces of code.

Reply:
Sure. Mine was just a comment that actually we _should_ rewrite the unrelated pieces of code. Anyway, all requested changes done.